shared/src/types/hash.rs

```diff
@@ -86,6 +88,26 @@ impl TryFrom<&[u8]> for Hash {
     }
 }
 
+impl TryFrom<String> for Hash {
+    type Error = self::Error;
+
+    fn try_from(string: String) -> HashResult<Self> {
+        let bytes: Vec<u8> =
+            Vec::from_hex(string).map_err(Error::FromStringError)?;
+        Self::try_from(bytes.as_slice())
+    }
+}
+
+impl TryFrom<&str> for Hash {
+    type Error = self::Error;
+
+    fn try_from(string: &str) -> HashResult<Self> {
+        let bytes: Vec<u8> =
+            Vec::from_hex(string).map_err(Error::FromStringError)?;
+        Self::try_from(bytes.as_slice())
```

Comment:
you could add the length check at this level, to avoid allocating a buffer with a potentially huge capacity. depending on where this is being used, it can be a DoS vector, crashing the ledger's process with the OOM killer or smth

Reply:
albeit to be honest, the source string is going to be twice as big

Reply:
I need to change this to `<[u8; 32]>::from_hex(..)`

Reply:
That's what I mentioned early that you reminded me of.